### DISCUSSION TITLE
clean up registered runners after helm uninstall automatically

### DISCUSSION THREAD

**Discussion by dongho-jung** [actions/actions-runner-controller#1698] [Questions] [answered]
with [post-delete hook](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) and GetRegistrationToken() below,

https://github.com/actions-runner-controller/actions-runner-controller/blob/bfc5ea4727f3dfb1a8cc590306f04f3d32133d4b/github/github.go#L149-L195

Can we clean up registered runners after helm uninstall automatically? I hope I can figure it out by myself though I don't have enough understanding this domain yet..

**Comment by dongho-jung** [comment]
On second thought,,, If ARC can auto-scaling then it should mean it can clean up scaled-in runners right? So I think it's easier than I thought before because the functionality I wish is already implemented somewhere. Or maybe helm already have supported this and didnt work for my environment.. I had an issue like [this](https://github.com/actions-runner-controller/actions-runner-controller/issues/1527)(failed calling webhook "mutate.runnerdeployment.actions.summerwind.dev ... lscontext deadline exceeded) and I've workarounded it by deleting blahblahwebhookconfigurations with [some cool guy's help](https://github.com/actions-runner-controller/actions-runner-controller/issues/1527). I have no idea why it works yet though, it works like a charm by the way. So I think those actions I did might break some hooks for cleaning up runners when uninstalling helm chart or maybe not.

**Comment by mumoshu** [comment, marked as answer]
The correct way to gracefully terminate/delete the runners is to delete the corresponding K8s custom resource, like `RunnerDeployment`.
So depending on your requirement, you'd need to trigger something similar to `kubectl delete runnerdeployment $RUNNER_DEPLOY_NAME` on every runner deployment in your cluster. If you'd need it to happen on helm uninstall(of ARC, not runners, right? Interesting), you'd need to write your own chart or umbrella chart to manage runners along with runners. It might work with or without hooks.

**Reply by dongho-jung** [reply]
Thank you it makes more sense! deleting CRD explicitly seems more flexible than hook : )
